### PR TITLE
Unicode logging fix

### PR DESCRIPTION
**Pull request by prodion23**
We were hitting some exceptions when attempting to log response/request bodys as part of the flask/requests instrumentation. 

Rather than logging the body at this level of instrumentation we will only log the request/response body when we get to the generic instrumentation(the main difference is we will only log after we have backslashescaped encoding errors and only if we are going to collect the response body)

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/hypertrace/agent/__init__.py [hunk 1]
```diff
@@ -70,6 +30,7 @@ def __new__(cls):
         if cls._instance is None:
             with cls._singleton_lock:
                 logger.debug('Creating Agent')
+                logger.debug('Python version: %s', sys.version)
```
Comment (prodion23) [inline src/hypertrace/agent/__init__.py:33]: this is just to help when trying to debug issues(sometimes the stack traces only include the `major.minor` version.

File: src/hypertrace/agent/__init__.py [hunk 2]
```diff
@@ -14,51 +14,11 @@
...
-        return logger_
-    except Exception as err:  # pylint: disable=W0703
-        print('Failed to customize logger: exception=%s, stacktrace=%s',
-              err,
-              traceback.format_exc())
-        return None
-
-
-# create logger object
-logger = setup_custom_logger(__name__)  # pylint: disable=C0103
```
Comment (prodion23) [inline src/hypertrace/agent/__init__.py:58]: This has just been moved to it's own file.

**Comment by jcchavezs** [comment]
My 2p here, I don't think agents should be logging this info. We should be smart about what we log as this kind of things seemed reasonable when we were in early stages of development but can be extremely verbose at this point of stability cc @davexroth

**Comment by jcchavezs** [comment]
Found this :p

image

cc @JBAhire

**Comment by prodion23** [comment]
I agree - I'm fine removing these logs entirely also, wasn't sure of historical context for them

**Pull request merged**